lib/proxy_fetcher/manager.rb

```diff
@@ -79,19 +91,33 @@ def get!
 
     # Clean current proxy list from dead proxies (that doesn't respond by timeout)
     def cleanup!
-      lock = Mutex.new
+      queue = Queue.new
+
+      proxies.dup.each do |proxy|
+        queue.push(proxy)
+      end
 
-      proxies.dup.each_slice(ProxyFetcher.config.pool_size) do |proxy_group|
-        threads = proxy_group.map do |group_proxy|
-          Thread.new(group_proxy, proxies) do |proxy, proxies|
-            lock.synchronize { proxies.delete(proxy) } unless proxy.connectable?
+      threads = []
+      lock = Mutex.new
+      workers = ProxyFetcher.config.pool_size
+      live_proxies = []
+
+      workers.times do
```

Comment:
Maybe we can decompose this method to something like:

1) `queue = build_proxies_queue`
2) `cleanup_queue!(queue)`

Or some sort of .. WDYT?

Reply:
I think that we can also create separate class to validate proxies list.
Something like `ProxyFetcher::ProxiesListValidator` which can reuse in the future.

Reply:
@yuryroot great idea! I love it